### ISSUE TITLE
[Nightly] Waypoint Export to notes, too many

### ISSUE THREAD

**Issue by geocachermgo** [cgeo/cgeo#8791]
**Describe the bug:**
I have 6 waypoints, when I store them in the notes I get 8.

**To Reproduce:**
Clear notes. Create Notes out of Waypoints.

**Actual behavior/state after performing these steps:**
For 6 waypoints there are 8 waypoint (see sceenshot)

**Expected behavior/state after performing these steps:**
For 6 waypoints there should be 6 waypoints

**Version of c:geo used:**
You will find the c:geo version in c:geo Menu -> About c:geo
2020.08.12-NB-a21754b
![Screenshot_2020-08-12-14-54-00-008_cgeo geocaching](https://user-images.githubusercontent.com/43813636/90018028-a17c8780-dcac-11ea-837d-67b62106bb6a.jpg)
![Screenshot_2020-08-12-14-50-54-487_cgeo geocaching](https://user-images.githubusercontent.com/43813636/90018083-b527ee00-dcac-11ea-8bfb-5b75e47923a2.jpg)

**Is the problem reproducible:**
Yes

**Screenshots:**
1) Waypoint on waypoint page
2) Waypoints in Note

**System information:**
```
--- System information ---
Device: MI 8 (dipper, Xiaomi)
Android version: 10
Android build: QKQ1.190828.002 test-keys
c:geo version: 2020.08.12-NB-a21754b
Google Play services: disabled - 20.26.14 (120400-320008519)
Low power mode: active
Compass capabilities: yes
Rotation vector sensor: present
Orientation sensor: present
Magnetometer & Accelerometer sensor: present
Direction sensor used: rotation vector
Hide caches: own/found 
Hide waypoints: -
HW acceleration: enabled (default state)
System language: de_DE
System date format: dd.MM.yy
Debug mode active: no
System internal c:geo dir: /data/user/0/cgeo.geocaching (89,6 GB free) internal
User storage c:geo dir: /storage/emulated/0/cgeo (89,6 GB free) external non-removable
Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (89,6 GB free) external non-removable
Database: /data/user/0/cgeo.geocaching/databases/data (10,1 MB) on system internal storage
Fine location permission: granted
Write external storage permission: granted
Geocaching sites enabled:
   geocaching.com: Logged in (Anmeldung OK) / PREMIUM
Geocaching.com date format: MM/dd/yyyy
Installed c:geo plugins:  none
--- End of system information ---

```

**Additional context:**
e.g. Reference to other issues, projects, sources, etc.

**Comment by eddiemuc** [comment]
Hi @geocachermgo , on your First Screenshot in the bottom it says "8 Wegpunkte". Are there really only 6 waypoints? Although I have to admit that the final and trailhead wps seem to be duplicated

**Comment by geocachermgo** [comment]
Yes, double checked it: had 6 waypoint (meanwile so were doubled, I guess be rereading from the notes).
By first "Wegpunkte spei..." (Store Waypoints) in the Note menue I got 6 waypoint in the notes. When I pressed it again I got 8 waypoints in the notes but still 6 in the cgeo waypoint list

**Comment by geocachermgo** [comment]
And thanks for the feature!

**Comment by geocachermgo** [comment]
Definition: cgeo-waypoint: waypoint in the swipe left cgeo
Note-waypoint: waypoint in the new new list in the notes.

I cleared the notes again, and then got 8 note-waypoints at the first click. Ok: there are 2 new cgeo-waypoint created but not shown in the cgeo waypoint list. If I close the cache, reopen again, then there are 8 cgeo waypoint.
I think that an older bug.
That leaves the question why the rereading creates new cgeo-waypoints.
Can you reproduce by adding this note, creating the cgeo-waypoints, deleting the notes and creating the cgeo-waypoint from that?
<----->
@Test WP 1 Ref (W) N46 16,050 E6 15,453
"Blabla"
@test WP 2 Stat mit Frage (X) N46 15,999 E6 15,426
"Hmhm"
@test WP 2 Trailhead (T) N46 16,050 E6 15,523
"Hmhm"
@test WP 2 Multi (S) N46 15,996 E6 15,523
"Hmhm"
@WP 2 Final (F) N46 15,955 E6 15,473
"Hmhm"
@WP 2 Par (P) N46 15,928 E6 15,473
"Hmhm"
</----->

**Comment by geocachermgo** [comment]
I deleted everything and manually created 6 cgeo-waypoints resulting in now 10 note-waypoints.
Type (O) seems strange?
I created the waypoints by "duplicate" the last waypoint changing type and setting the coords to the cache and shifted by 60°,120°,... and 100m.
The last 4 are double
<----->
@Referenzpunkt 1 (W) N46 16,049 E6 15,510
@p 1 als Kopie (P) N46 15,995 E6 15,510
@Trailhead (T) N46 15,968 E6 15,443
@Multi (S) N46 15,995 E6 15,376
@Final (F) N46 16,049 E6 15,376
@Frage (X) N46 16,076 E6 15,443
@Referenzpunkt 1 (O) N46 16,049 E6 15,510
@p 1 als Kopie (P) N46 15,995 E6 15,510
@Multi (S) N46 15,995 E6 15,376
@Final (F) N46 16,049 E6 15,376
</----->

**Comment by eddiemuc** [comment]
OK, I think I understand the issue now. I suspect this has to do with the way coordinates are compared with each other (coordinates are Sometimes considered different even if they are equal uo to the thousand-second).
I will definitely Look into this, but can only do this Starring next monday (no Laptop until then)

Thanks for the Report!

**Comment by MagpieFourtyTwo** [comment]
> (coordinates are Sometimes considered different even if they are equal uo to the thousand-second).

Same same. BTW, with projections: The result of c:geo's calculation after entering distance and bearing (in WP's properties) is in most of the cases considered to *not* be equal to the coordinates you afterwards put into the PN (for logging purposes). That's why c:geo afterwards will create a seemingly duplicate waypoint in the WP list (automatically from the PN).

Already talked about this with @moving-bits and AFAIR he had an approach to eliminate these rounding errors resp. the discrepancy between internal value and external representation. Perhaps he still has s/th in mind ...?

**Comment by moving-bits** [comment]
> Already talked about this with @moving-bits and AFAIR he had an approach to eliminate these rounding errors resp. the discrepancy between internal value and external representation. Perhaps he still has s/th in mind ...?

Can't remember. Are you thinking of #8314? @UniQP has assigned that issue to himself.

**Comment by MagpieFourtyTwo** [comment]
Hmm, maybe. Although I had an older discussion in mind. 
But doesn't matter anymore, as you exactly redirected to the correct stuff.
So my approach via mentioning you worked perfectly fine.  ;)
Thx a lot for the clarification!

**Comment by eddiemuc** [comment]
@MagpieFourtyTwo , @moving-bits 
I would approach this by changing the compared Logic for waypoint coordinates from "compare pure lat/lon" to "compare coordinate representation in Standard DD MM.MMM Format". This might solve also the projection Problem

@geocachermgo :
Would you be so kind and look up for a pair of duplicated waypoints whether the coordinates differ when you Change Display to DD.DDDD°? If yes, this would confirm my suspicion

**Comment by geocachermgo** [comment]
Yes, I can confirm that the coords in dd.dddd display ate different by typically 6e-6 in the longitude
![Screenshot_2020-08-13-11-19-04-353_cgeo geocaching](https://user-images.githubusercontent.com/43813636/90117247-f07ef700-dd56-11ea-8e1f-1e56f47b3829.jpg)

**Comment by geocachermgo** [comment]
Btw: is the an option to change the format of the coordinate globally in cgeo?

**Comment by MagpieFourtyTwo** [comment]
> Btw: is the an option to change the format of the coordinate globally in cgeo?

Why should someone want to do so? As on gc.com DD MM.MMM is kind of standard, changing it in c:geo.only will most probably lead to complete coordinate confusion ...

**Comment by geocachermgo** [comment]
Was that a "no" to that question? It was not a request to have such an option but the question if its possible.

**Comment by geocachermgo** [comment]
Intermediate summary: I see 3 issues:
1) Identification of close-by coodinates
2) Type (O) of created note-waypoint
3) Update of waypoint list when not reloading the cache (6 in cgeo-waypoint list, 8 in bottom display and created note-waypoints) (connected to #8237 ? Which is also an waypoint update problem?)

**Comment by MagpieFourtyTwo** [comment]
> Was that a "no" to that question? 

There at least is the option to change input format, but I have not tested what it does to representation in the rest.

**Comment by eddiemuc** [comment]
> 2) Type (O) of created note-waypoint

There is (in Code at least) a waypoint Type "Own" whose shortid is O. Apart from the Name I think it is handled much the same as reference Points (e.g. same Symbol, same Default Name). Can't tell you right now why this exists as a separate Type and when it is created (maybe when bearing is used)?
![Screenshot_20200814-081456.jpg](https://user-images.githubusercontent.com/6909759/90219682-e5839f80-de06-11ea-9e17-229e39b4ebaa.jpg)

**Comment by eddiemuc** [comment]
> 1) Identification of close-by coodinates
> 3) Update of waypoint list when not reloading the cache (6 in cgeo-waypoint list, 8 in bottom display and created note-waypoints) 

Those two should be solvable by changing the compare logic for waypoints coordinates when parsing PN. Will do that as soon as I can.

**Comment by eddiemuc** [comment]
> > 1. Identification of close-by coodinates
> > 2. Update of waypoint list when not reloading the cache (6 in cgeo-waypoint list, 8 in bottom display and created note-waypoints)
> 
> Those two should be solvable by changing the compare logic for waypoints coordinates when parsing PN. Will do that as soon as I can.

These two should be solved with #8817. 

As far as waypoint type O/OWN is concerned: I couldn't find a place in code where this type is still assigned, it seems like a "legacy type" to me which maybe was used in earlier versions of c:geo. Could this be a "left-over"?
